fineract-provider/src/main/java/org/apache/fineract/batch/exception/ErrorHandler.java

```diff
@@ -99,13 +99,13 @@ public static ErrorInfo handler(final RuntimeException exception) {
             return handleException(e, new PlatformDomainRuleExceptionMapper(), 9999);
         }
         if (exception instanceof TransactionException) {
-            return new ErrorInfo(HttpStatus.SC_BAD_REQUEST, 4001, "{\"Exception\": %s}".formatted(exception.getMessage()));
+            return new ErrorInfo(HttpStatus.SC_INTERNAL_SERVER_ERROR, 4001, "{\"Exception\": %s}".formatted(exception.getMessage()));
```

Comment:
if you are changing this for 5xx error, you should change the code as well (4001 -> 5001?)